### PR TITLE
Moving and Node base classes

### PR DESCRIPTION
**Pull request by icgood**
Moved common functionality into base classes:

* `Node`: represents something we put on the screen, so it has an image, location, and rotation
* `Moving`: A Node that has a velocity and can rotate or accelerate, with `onLeave` callbacks when the image moves off the screen

I also added `constants.dtCoefficient` that creates a single value that scales how fast the game goes. Every use of `dt` should be multiplied by this coefficient.

### CODE REVIEW

**Comment by abtrout** [comment]
I LOVE IT

File: app.js [hunk 1]
```diff
@@ -21,8 +25,11 @@ function bindKeys() {
...
 }
 
+/**
+ * Initializes the game state and binds the tick function to run
+ * every time the screen is rendered.
+ */
 Stage(function(stage) {
   let state = new State(bindKeys())
   state.start(stage)
-  stage.tick(dt => state.tick(dt, stage))
```
Comment (icgood) [inline app.js]: I removed this because now nodes (like shuttle) register their own tick functions.

File: lib/node.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    // Set the node's pinned location and rotation values
+    this.node.pin({
+      align: 0.5,
+      offsetX: this.offsetX,
+      offsetY: this.offsetY,
+      rotation: this.rotation,
+    })
+
+    // Register the tick method
+    this.node.tick(dt => this.tick(dt, parentNode))
```
Comment (icgood) [inline lib/node.js]: Now every time `foo.start(stage)` is called, it registers its own tick function.

File: lib/bullet.js [hunk 3]
```diff
@@ -1,41 +1,36 @@
 import { constants } from './constants.js'
+import { Moving } from './moving.js'
 
-export class Bullet {
+export class Bullet extends Moving {
```
Comment (abtrout) [inline lib/bullet.js]: it's a laser, dude

Comment (icgood) [reply lib/bullet.js]: does it pew?

Comment (icgood) [reply lib/bullet.js]: does it pew?

**Pull request merged**